administrator/components/com_media/controllers/file.json.php

```diff
@@ -82,10 +82,10 @@ public function upload()
 
 			// Transform filename to punycode
 			$fileparts['filename'] = JStringPunycode::toPunycode($fileparts['filename']);
-			$tempExt = (!empty($fileparts['extension'])) ? strtolower($fileparts['extension']) : '';
+			$tempExt = !empty($fileparts['extension']) ? strtolower($fileparts['extension']) : '';
 
 			// Transform filename to punycode, then neglect otherthan non-alphanumeric characters & underscores. Also transform extension to lowercase
-			$safeFileName = preg_replace(array("/[\\s]/", '/[^a-zA-Z0-9_]/'), array('_', ''), $fileparts['filename']) . '.' . $tempExt;
+			$safeFileName = preg_replace(array("/[\\s]/", '/[^a-zA-Z0-9_\-]/'), array('_', ''), $fileparts['filename']) . '.' . $tempExt;
```

Comment:
```suggestion
			$safeFileName = preg_replace(array("/[\\s]/", '/[^a-zA-Z0-9_-]/'), array('_', ''), $fileparts['filename']) . '.' . $tempExt;
```
No need to escape the hyphen.